### PR TITLE
vk: async submission

### PR DESCRIPTION
**Pull request by YunHsiao**

### CODE REVIEW

File: cocos/renderer/gfx-vulkan/VKGPUObjects.h [hunk 1]
```diff
@@ -470,6 +474,218 @@ class CCVKGPUCommandBufferPool : public Object {
...
+            if (cur->size - cur->curOffset > gpuBuffer->size) {
+                buffer = cur;
+                break;
+            }
+        }
+        if (!buffer) {
+            _pool.resize(bufferCount + 1);
+            buffer = &_pool.back();
+            VkBufferCreateInfo bufferInfo{VK_STRUCTURE_TYPE_BUFFER_CREATE_INFO};
+            bufferInfo.size = buffer->size;
```
Comment (Mee-gu) [inline cocos/renderer/gfx-vulkan/VKGPUObjects.h:513]: should size be gpuBuffer->size?

File: cocos/renderer/gfx-vulkan/VKGPUObjects.h [hunk 2]
```diff
@@ -470,6 +474,218 @@ class CCVKGPUCommandBufferPool : public Object {
     vector<uint> _counts;
 };
 
+class CCVKGPUStagingBufferPool : public Object {
+public:
+    struct Buffer {
```
Comment (Mee-gu) [inline cocos/renderer/gfx-vulkan/VKGPUObjects.h:479]: struct Buffer 可以放到 private 里

File: cocos/renderer/gfx-vulkan/VKGPUObjects.h [hunk 3]
```diff
@@ -470,6 +474,218 @@ class CCVKGPUCommandBufferPool : public Object {
...
+    }
+
+private:
+    CCVKGPUDevice *_device = nullptr;
+    vector<Buffer> _pool;
+};
+
+class CCVKGPURecycleBin : public Object {
+public:
+    struct Buffer {
```
Comment (Mee-gu) [inline cocos/renderer/gfx-vulkan/VKGPUObjects.h:541]: 可以放在 private 里，下同

File: cocos/renderer/gfx-gles2/GLES2Texture.cc [hunk 4]
```diff
@@ -21,8 +21,8 @@ bool GLES2Texture::initialize(const TextureInfo &info) {
     _width = info.width;
     _height = info.height;
     _depth = info.depth;
-    _arrayLayer = info.arrayLayer;
-    _mipLevel = info.mipLevel;
+    _layerCount = info.arrayLayer;
+    _levelCount = info.mipLevel;
```
Comment (minggo) [inline cocos/renderer/gfx-gles2/GLES2Texture.cc:25]: Info 里的名字最好也统一一下吧。

File: cocos/renderer/gfx-vulkan/VKGPUObjects.h (hunk 1 above)
Comment (YunHsiao) [reply cocos/renderer/gfx-vulkan/VKGPUObjects.h:513]: nope, the staging buffer is allocated in chunks, 16M per chunk for now

**Pull request merged**